hw/ip/pwrmgr/data/pwrmgr_testplan.hjson

```diff
@@ -11,22 +11,225 @@
     {
       name: smoke
       desc: '''
-            Smoke test accessing a major datapath within the pwrmgr.
+            Smoke test exercising the pwrmgr state transitions.
+
+            - Brings pwrmgr out of POR.
+            - Enables wakeup.
+            - Triggers SW initiated low power transition.
+            - Triggers wakeup.
+            - Enables reset.
+            - Triggers a reset.
+            - Waits for pwrmgr to be out of reset.
 
             **Stimulus**:
-            - TBD
+            - CSR writes to `wakeup_en`, `reset_en`, and `low_power_hint`.
+            - Needs many input pins to line up correctly in order to prevent the
+              pwrmgr from waiting forever.
 
             **Checks**:
-            - TBD
+            - Checks that fast fsm is active checking the CSR `ctrl_cfg_regwen`
+              is set.
+            - Checks that the wakeup and reset causes are as expected reading
+              CSRs `wake_status` and `reset_status`.
+            - Checks the output `pwr_rst_req.reset_cause` matches a low power or
+              reset cause.
+            - Checks the output `pwr_rst_req.rstreqs` matches the enabled
+              resets.
             '''
       milestone: V1
       tests: ["pwrmgr_smoke"]
     }
     {
-      name: feature1
-      desc: '''Add more test entries here like above.'''
-      milestone: V1
+      name: wakeup
+      desc: '''
+            Test random wakeup, wakeup_en, wake_info_capture_dis, and
```

Comment:
one thing to note here, is that since there can be multiple wakeups, and they can all happen at the same time, we want to make sure we record ALL the wakeup reasons that happen during the wakeup window. It would be great to stress test this to make sure we don't miss anything.

Reply:
Yes, the intention is to set all bits of wakeups_i at random, not as one-hot.